edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/MetricsModule.cs

```diff
@@ -18,18 +19,25 @@ public sealed class MetricsModule : Module
         MetricsConfig metricsConfig;
         string iothubHostname;
         string deviceId;
+        string edgeAgentStorageFolder;
 
-        public MetricsModule(MetricsConfig metricsConfig, string iothubHostname, string deviceId)
+        public MetricsModule(MetricsConfig metricsConfig, string iothubHostname, string deviceId, string edgeAgentStorageFolder)
         {
             this.metricsConfig = Preconditions.CheckNotNull(metricsConfig, nameof(metricsConfig));
             this.iothubHostname = Preconditions.CheckNotNull(iothubHostname, nameof(iothubHostname));
             this.deviceId = Preconditions.CheckNotNull(deviceId, nameof(deviceId));
+            if (!Directory.Exists(Preconditions.CheckNotNull(edgeAgentStorageFolder, nameof(edgeAgentStorageFolder))))
```

Comment:
nit: split this into 2 lines

Reply:
this is existing code issue; iothuhHostname and deviceId should call Preconditions.CheckNullOrWhiteSpace.